src/estimagic/optimization/trust_region_management.py

```diff
@@ -0,0 +1,344 @@
+        n_points (int): Number of points in the trust region at which criterion values
+            are known. The actual number can be larger than this if the existing points
+            are badly spaced.
+        existing_points (np.ndarray): 2d Array where each row is a parameter vector at
+            which the criterion has already been evaluated.
+        optimality_criterion (str): One of "a-optimal", "d-optimal", "e-optimal",
+            "g-optimal" or "maximin". Default "maximin".
+        lhs_design (str): One of "random", "centered". Determines how sample points are
+            spaced inside bins. Default 'centered'.
+        target (str): One of "linear" or "quadratic". Determines in which space the
+            criterion is evaluated. For "quadratic" the Latin-Hypercube sample X is
+            mapped onto Y(X) = [X, X ** 2] and the criterion value for X is computed
+            using Y(X).  This can be useful if one chooses an optimality criterion that
+            minimizes e.g.  the variance of the least-squares estimator, while using a
+            quadratic or polynomial model. Default is "linear".
+        n_iter (int): Iterations considered in random search.
+
+    Returns:
+        out (dict): Dictionary with entries:
+        - 'points' (np.ndarray): The (optimal) Latin hypercube sample. Has shape
+        (n_points, len(center)).
+        - 'crit_vals' (np.ndarray): 1d array of length n_iter, containing the criterion
+        values assigned to each candidate sample. Mainly used for debugging.
+
+    """
+    if lhs_design not in {"centered", "random"}:
+        raise ValueError(
+            "Invalid Latin hypercube design. Must be in {'random', 'centered'}"
+        )
+
+    n_dim = len(center)
+    dtype = np.uint8 if n_points < 256 else np.uint16
+
+    if existing_points is None:
+        candidates = _create_upscaled_lhs_sample(n_dim, n_points, n_iter, dtype)
+        n_new_points = n_points
+    else:
+        existing_upscaled = _scale_up_points(existing_points, center, radius, n_points)
+        empty_bins = _get_empty_bin_info(existing_upscaled, n_points)
+        candidates = _extend_upscaled_lhs_sample(empty_bins, n_points, n_iter, dtype)
+        existing_upscaled = np.tile(existing_upscaled, (n_iter, 1, 1))
+        candidates = np.concatenate((existing_upscaled, candidates), axis=1)
+        n_new_points = n_points - len(existing_points)
+
+    candidates = candidates.astype(float)
+    if lhs_design == "centered" and n_new_points > 0:
+        candidates[:, -n_new_points:, :] += 0.5
+    elif lhs_design == "random" and n_new_points > 0:
+        candidates[:, -n_new_points:, :] += np.random.uniform(
+            size=(n_new_points, n_dim)
+        )
+
+    candidates = _scale_down_points(candidates, center, radius, n_points)
+
+    crit_func = partial(
+        compute_optimality_criterion, criterion=optimality_criterion, target=target
+    )
+    crit_vals = crit_func(candidates)
```

Comment:
Why partial if we just do one function call afterwards and could also pass in all arguments in the call?

Reply:
Yes, I agree, it is unnecessary.